### ISSUE TITLE
Feature request: auto surround quotes

### ISSUE THREAD

**Issue by Kuroonehalf** [focus-editor/focus#444]
Unsure if the default behavior in other editors applies to single tick quotes, since it could interfere with apostrophes in strings, but I figure this should be easy enough to add for double tick quotes. I write game scripts in a python language so I use quotes constantly, and quote auto surround is a nice QOL feature.

**Comment by focus-editor** [comment]
How is this different from the existing auto surround feature, please
clarify

**Comment by Kuroonehalf** [comment]
I assume you mean the `auto_surround_with_brackets_and_quotes` option. Even with it toggled on, I don't seem to get any quote auto surround. Is it supposed to work?

**Comment by focus-editor** [comment]
It should auto-surround any selection when you type a quote char.

**Comment by Kuroonehalf** [comment]
Oh, I understand now. You make a selection, then press quote. I mean I'd like it to work without selection, the way auto surround brackets/braces work.

**Comment by focus-editor** [comment]
You mean you type a quote and it types another quote?

**Comment by Kuroonehalf** [comment]
Right. Types another quote, 1 character ahead of the cursor. The basic implemention of this seems like it might be that simple. 

A more fancy implementation might want to detect when the quote is opening or closing a string, and if it's closing, don't type the extra quote. It should also account for deleting an empty quote string (i.e.  [quote][cursor][quote]), which ideally should delete both quotes.

I forgot to mention, but this feature is also nice so that when you're typing a string, it doesn't interpret it as code and syntax highlight it incorrectly.

**Comment by focus-editor** [comment]
While this feature is something some people will definitely find useful, I
believe it crosses the line between usefulness and annoyingly staying in
your way. It wasn't implemented deliberately.
The solution is to type two quotes and then press back (2 keystrokes), and
this way you're in control in 100% of the cases, whereas with that feature
there will always be situations where it types a quote you didn't want. We
don't want this even for an opt-in feature.

**Comment by Kuroonehalf** [comment]
I'm not sure I understand this possible annoying scenario. When would the added quote get in the way? Generally, all quotes opened must be closed, no? You open it, type whatever you want to type, then press right to go to the right of the closing quote. Unless you're single tick quoting a double tick quote, but that strikes me as an exceedingly rare scenario. On the other hand, any random game script will have thousands of lines in quotes, and a little QOL feature like this would go a long way. I guess if you're really opposed I'll drop my case, but it's a feature that would be useful to many.

**Comment by focus-editor** [comment]
Typing 3 quotes for a docstring in python.
Typing an opening quote on one line and a closing quote on another line.
Deleting a quote in the middle of other quotes.
Using multiple cursors to type quotes, some of which might have a quote
already next to them.

These are things which have personally annoyed me in the past that came to
mind.

Focus is about always being fully in control of what appears on screen, and
with this feature it's hard to impossible to get every case right, so we
choose not to implement it at all.

You could bring it up on discord to gauge public support for this feature.
If everyone overwhelmingly wants it we might reconsider.
